### PR TITLE
test: pipeline-api-tests

### PR DESCRIPTION
**Pull request by neha00290**
## Summary by CodeRabbit

- **Tests**
	- Introduced a new end-to-end test for pipeline creation, enabling, and disabling.
	- Utilized parameterization to test various scenarios with different pipeline names, actions, and expected HTTP status codes.
	- Removed the previous end-to-end tests related to user token management, which included user token retrieval and email verification processes.

### CODE REVIEW

File: tests/api-testing/tests/test_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        ],
+        "edges": [
+            {
+                "id": "e1e1fa129-cfbd-4082-8d39-c4dc418a670f-8e0ed123-7737-4801-9466-bd909c425c72",
+                "source": "1e1fa129-cfbd-4082-8d39-c4dc418a670f",
+                "target": "8e0ed123-7737-4801-9466-bd909c425c72",
+            }
+        ],
+        "org": org_id,
+    }
```
[the remark this answers is not in the document]
Comment (neha00290) [reply tests/api-testing/tests/test_pipeline.py:61]: added

**Review by YashodhanJoshi1 — APPROVED** [review]

File: tests/api-testing/tests/test_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        print(f"Pipeline {pipeline_id} disabled successfully.")
+
+     
+    resp_get_pipeline = session.get(f"{url}api/{org_id}/pipelines")
+    print(resp_get_pipeline.json())
+    assert resp_get_pipeline.status_code == 200
+    if action == "enable":
+       assert resp_get_pipeline.json()["list"][0]["enabled"] is True
+    if action == "disable":
+       assert resp_get_pipeline.json()["list"][0]["enabled"] is False
```
[the remark this answers is not in the document]
Comment (neha00290) [reply tests/api-testing/tests/test_pipeline.py]: added

File: tests/api-testing/tests/test_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import json
+import requests
+import pytest
+from datetime import datetime, timezone, timedelta
+import time
+
```
[the remark this answers is not in the document]
Comment (neha00290) [reply tests/api-testing/tests/test_pipeline.py]: removed

File: tests/api-testing/tests/test_pipeline.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    assert resp_get_pipeline.status_code == 200
+    if action == "enable":
+       assert resp_get_pipeline.json()["list"][0]["enabled"] is True
+    if action == "disable":
+       assert resp_get_pipeline.json()["list"][0]["enabled"] is False
+
+    # Delete the pipeline
+    resp_delete_pipeline = session.delete(f"{url}api/{org_id}/pipelines/{pipeline_id}")
+    assert resp_delete_pipeline.status_code == 200, f"Expected status code 200 but got {resp_delete_pipeline.status_code}"
+    print(f"Pipeline {pipeline_id} deleted successfully.")
```
[the remark this answers is not in the document]
Comment (neha00290) [reply tests/api-testing/tests/test_pipeline.py]: added

File: tests/api-testing/tests/test_pipeline.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+                "io_type": "output",
+            },
+        ],
+        "edges": [
+            {
+                "id": "e1e1fa129-cfbd-4082-8d39-c4dc418a670f-8e0ed123-7737-4801-9466-bd909c425c72",
+                "source": "1e1fa129-cfbd-4082-8d39-c4dc418a670f",
+                "target": "8e0ed123-7737-4801-9466-bd909c425c72",
+            }
+        ],
```
[the remark this answers is not in the document]
Comment (neha00290) [reply tests/api-testing/tests/test_pipeline.py:54]: this is needed

File: tests/api-testing/tests/test_pipeline.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        "org": org_id,
+    }
+
+    # Create a pipeline
+    resp_create_pipeline = session.post(f"{url}api/{org_id}/pipelines", json=pipeline_payload)
+    assert resp_create_pipeline.status_code == expected_status, f"Expected status code {expected_status} but got {resp_create_pipeline.status_code}"
+    print("Create pipeline response:", resp_create_pipeline.json())
```
[the remark this answers is not in the document]
Comment (neha00290) [reply tests/api-testing/tests/test_pipeline.py:61]: keeping this for now

File: tests/api-testing/tests/test_pipeline.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+    # Retrieve the pipeline list
+    resp_list_pipelines = session.get(f"{url}api/{org_id}/pipelines")
+    assert resp_list_pipelines.status_code == 200, f"Expected status code 200 but got {resp_list_pipelines.status_code}"
+    
+    # Parse the pipeline list response
+    pipelines_list = resp_list_pipelines.json().get("list", [])
+    assert pipelines_list, "No pipelines found in the list response."
```
[the remark this answers is not in the document]
Comment (neha00290) [reply tests/api-testing/tests/test_pipeline.py:69]: this is added

**Pull request merged**